### ISSUE TITLE
onStateChanged not triggered when iOS is in lock screen

### ISSUE THREAD

**Issue by hellohejinyu** [llfbandit/record#121]
**Package version**
4.4.3

**Describe the bug**
The recording was paused in the iOS lock screen state (maybe it was automatically paused by the system), but onStateChanged was not triggered.

**To Reproduce**
1. start recording in iOS 16.1.1
2. lock the screen
3. at this point the recording is paused and can be resumed automatically after unlocking
4. but there is no output from onStateChange during the whole process

**Expected behavior**
onStateChange works fine so that it stops the recording time tick when the screen is locked

**Smartphone (please complete the following information):**
 - Device: iPhone 12 Pro
 - OS: iOS 16.1.1

**Comment by llfbandit** [comment]
You're right, the recording process is paused by the system.
Unfortunately, as far as I know, there's no event we could catch to handle this case internally at the moment.

**Comment by llfbandit** [comment]
I may have found a way to fix this. This is available on v5 branch. Some breaking changes are on the way that can be quickly fixed.
I don't have an iPhone to confirm the changes. Are you available for some testing?

Thanks

**Comment by hellohejinyu** [comment]
throw an platform exception

```console
flutter: pcm16bit supported: true
flutter: [      id: com.apple.avfoundation.avcapturedevice.built-in_audio:0
flutter:       label: iPhone 麦克风
flutter:       channels: null
flutter:       samplingRate: null
flutter:       ]
flutter: PlatformException(record, Failed to start recording, The operation couldn’t be completed. (OSStatus error -50.), null)
```

https://github.com/llfbandit/record/assets/8766034/67d5360e-2912-4398-a9df-5ee7e4d6af0c

**Comment by llfbandit** [comment]
Available in v5-beta.1.

**Comment by llfbandit** [comment]
Released in record_darwin 1.0.1
